app/src/main/java/de/westnordost/streetcomplete/quests/barrier_bicycle_barrier_type/AddBicycleBarrierType.kt

```diff
@@ -0,0 +1,31 @@
+package de.westnordost.streetcomplete.quests.barrier_bicycle_barrier_type
+
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.edits.update_tags.StringMapChangesBuilder
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmFilterQuestType
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.BICYCLIST
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.BLIND
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.CAR
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.PEDESTRIAN
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.WHEELCHAIR
+
+class AddBicycleBarrierType : OsmFilterQuestType<BicycleBarrierType>() {
+
+    override val elementFilter = """
+        nodes with barrier=cycle_barrier and !cycle_barrier
+    """
+    override val commitMessage = "Add specific cycle barrier type"
+    override val wikiLink = "Key:cycle_barrier"
+    override val icon = R.drawable.ic_quest_barrier // TODO
+    override val isDeleteElementEnabled = true
+
+    override val questTypeAchievements = listOf(CAR, PEDESTRIAN, BLIND, WHEELCHAIR, BICYCLIST)
+
+    override fun getTitle(tags: Map<String, String>) = R.string.quest_bicycle_barrier_type_title
+
+    override fun createForm() = AddBicycleBarrierTypeForm()
+
+    override fun applyAnswerTo(answer: BicycleBarrierType, changes: StringMapChangesBuilder) {
+        changes.modify("cycle_barrier", answer.osmValue)
```

Comment:
Does `modify` work here, when the element filter explicitly excludes elements with `cycle_barrier`?

```suggestion
        changes.add("cycle_barrier", answer.osmValue)
```